packages/@storefront/sayt/src/previous-search-terms/index.ts

```diff
@@ -0,0 +1,37 @@
+import { provide, tag, Events, Selectors, Store, Tag } from '@storefront/core';
+import Sayt from '../sayt';
+
+@provide('previousSearch', (props) => props)
+@tag('gb-sayt-previous-search', require('./index.html'))
+class PreviousSearch {
+  props: PreviousSearch.Props = {
+    onClick: (query) => () => this.actions.search(query),
+  } as any;
+
+  constructor() {
+    const previousSearches = []
+    this.state = {...this.state, previousSearches}
+  }
+
+  init() {
+    this.flux.on(Events.ORIGINAL_QUERY_UPDATED, this.updatePreviousSearches);
+  }
+
+  updatePreviousSearches = (originalQuery: string) => {
+    if(!this.state.previousSearches.includes(originalQuery) && this.state.previousSearches.length < 6) {
```

Comment:
Lets split up the expressions into two separate conditionals:

ie.

```
IF ( does not include query )
    ..
  IF ( has less than max )
    ..
  ELSE
    ..
```

Reply:
Also, please add a space after keywords (`if`, `for`, `while`, etc.).